src/tera.rs

```diff
@@ -390,6 +390,28 @@ impl Tera {
         }
     }
 
+    /// Returns an iterator over the names of all registered templates in an
+    /// unspecified order.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tera::Tera;
+    ///
+    /// let mut tera = Tera::default();
+    /// tera.add_raw_template("foo", "{{ hello }}");
+    /// tera.add_raw_template("another-one.html", "contents go here");
+    ///
+    /// let names: Vec<_> = tera.get_template_names().collect();
+    /// assert_eq!(names.len(), 2);
+    /// assert!(names.contains(&"foo"));
+    /// assert!(names.contains(&"another-one.html"));
+    /// ```
+    #[inline]
+    pub fn get_template_names(&self) -> impl Iterator<Item = &str> {
+        self.templates.keys().map(|s| s.as_str())
```

Comment:
I think it would be worth `collect()`ing in the function as most people will do it anyway and it's not a function that will be called so many times the allocation would be detrimental

Reply:
Why `collect()` in the function when the user can do this themselves? In general, you shouldn't allocate for your users. They may want a `Vec`, or they may want a `HashSet`, or they may simply want the `.count()`.